controller/eks-cluster-config-handler.go

```diff
@@ -178,6 +178,17 @@ func (h *Handler) OnEksConfigRemoved(_ string, config *eksv1.EKSClusterConfig) (
 		logrus.Infof("waiting for config [%s] node groups to delete", config.Name)
 	}
 
+	for _, ng := range config.Spec.NodeGroups {
+		if aws.StringValue(ng.NodeRole) == "" {
+			logrus.Infof("deleting node instance role for config [%s]", config.Name)
+			err = deleteStack(svc, fmt.Sprintf("%s-node-instance-role", config.Spec.DisplayName), fmt.Sprintf("%s-node-instance-role", config.Spec.DisplayName))
+			if err != nil {
+				return config, fmt.Errorf("error deleting worker node stack: %v", err)
+			}
+			break
+		}
+	}
+
```

Comment:
Let's keep this where it was, but add your changes.

Reply:
Let me know if was not resolved as should be.